crates/astria-composer/src/metrics_init.rs

```diff
@@ -0,0 +1,234 @@
+        "The number of transactions successfully sent by the grpc collector before sending it to \
+         be bundled"
+    );
+
+    // executor metrics
+    describe_counter!(
+        TRANSACTIONS_RECEIVED,
+        Unit::Count,
+        "The number of transactions successfully received from collectors and bundled labelled by \
+         rollup"
+    );
+    describe_counter!(
+        TRANSACTIONS_DROPPED_TOO_LARGE,
+        Unit::Count,
+        "The number of transactions dropped because they were too large"
+    );
+    describe_counter!(
+        NONCE_FETCH_COUNT,
+        Unit::Count,
+        "The number of times we have attempted to fetch the nonce"
+    );
+    describe_counter!(
+        NONCE_FETCH_FAILURE_COUNT,
+        Unit::Count,
+        "The number of times we have failed to fetch the nonce"
+    );
+    describe_histogram!(
+        NONCE_FETCH_LATENCY,
+        Unit::Milliseconds,
+        "The latency of nonce fetch"
+    );
+    describe_gauge!(CURRENT_NONCE, Unit::Count, "The current nonce");
+    describe_histogram!(
+        TRANSACTION_SUBMISSION_LATENCY,
+        Unit::Milliseconds,
+        "The latency of submitting a transaction to the sequencer"
+    );
+    describe_counter!(
+        TRANSACTION_SUBMISSION_FAILURE_COUNT,
+        Unit::Count,
+        "The number of failed transaction submissions to the sequencer"
+    );
+    describe_counter!(
+        BUNDLES_SUBMISSION_SUCCESS_COUNT,
+        Unit::Count,
+        "The number of successful bundle submissions to the sequencer"
+    );
+    describe_counter!(
+        BUNDLES_SUBMISSION_FAILURE_COUNT,
+        Unit::Count,
+        "The number of failed bundle submissions to the sequencer"
+    );
+    describe_histogram!(
+        BUNDLES_OUTGOING_TRANSACTIONS_COUNT,
+        Unit::Count,
+        "The number of outgoing transactions to the sequencer"
+    );
+    describe_histogram!(
+        BUNDLES_OUTGOING_BYTES,
+        Unit::Bytes,
+        "The size of bundles in the outgoing bundle"
+    );
+    describe_counter!(
+        BUNDLES_NOT_DRAINED,
+        Unit::Count,
+        "The number of bundles not drained during shutdown"
+    );
+    describe_counter!(
+        BUNDLES_DRAINED,
+        Unit::Count,
+        "The number of bundles drained successfully during shutdown"
+    );
+
+    // bundle factory metrics
+    describe_counter!(
+        BUNDLES_TOTAL_COUNT,
+        Unit::Count,
+        "The total number of finished bundles constructed over the composer's lifetime"
+    );
+    describe_histogram!(
+        BUNDLES_TOTAL_BYTES,
+        Unit::Bytes,
+        "The distribution of sizes of finished bundles constructed over the composer's lifetime"
+    );
+    describe_histogram!(
+        BUNDLES_TOTAL_TRANSACTIONS_COUNT,
+        Unit::Count,
+        "The total number of transactions in finished bundles constructed over the composer's \
+         lifetime"
+    );
+}
+
+pub const GETH_COLLECTOR_TRANSACTIONS_COLLECTED: &str = concat!(
```

Comment:
IMO this should be `TRANSACTIONS_COLLECTED`, but potentially distinguished with a label.